clanker-deployer/DeployToken.ts

```diff
@@ -0,0 +1,407 @@
+        .map((b) => b.toString(16).padStart(2, '0'))
+        .join('')
+  );
+  let i = startingPoint;
+  const salt = `0x${i.toString(16).padStart(64, '0')}` as `0x${string}`;
+
+  const token = await predictToken_v3_1(
+    admin,
+    name,
+    symbol,
+    image,
+    metadata,
+    context,
+    supply,
+    salt,
+    initialSupplyChainId
+  );
+
+  return { salt, token };
+}
+
+// Replace the dummy transaction sender with a real implementation
+async function sendTransactionAsync({
+  to,
+  data,
+  value,
+  gas,
+}: {
+  to: `0x${string}`;
+  data: `0x${string}`;
+  value: bigint;
+  gas: bigint;
+}): Promise<string> {
+  const privateKey = process.env.PRIVATE_KEY as `0x${string}`;
+  if (!privateKey) {
+    throw new Error('PRIVATE_KEY environment variable is not set');
+  }
+
+  const account = privateKeyToAccount(privateKey);
+  const client = createWalletClient({
+    account,
+    chain: base,
+    transport: http(process.env.RPC_URL || 'https://mainnet.base.org'),
+  });
+
+  const hash = await client.sendTransaction({
+    to,
+    data,
+    value,
+    gas,
+    chain: base,
+  });
+
+  return hash;
+}
+
+/**
+ * Builds a token deployment transaction
+ */
+async function buildDeploymentTransaction({
+  deployerAddress,
+  formData,
+  chainId,
+  userInfo,
+  desiredPrice = 10, // Default desired price (pairs with WETH)
+}: {
+  deployerAddress: `0x${string}`;
+  formData: IDeployFormData;
+  chainId: number;
+  userInfo?: IUserInfo;
+  desiredPrice?: number;
+}): Promise<{
+  transaction: {
+    to: `0x${string}`;
+    data: `0x${string}`;
+  };
+  value: bigint;
+}> {
+  if (!deployerAddress) {
+    throw new Error('No account address found');
+  }
+
+  // Create metadata and social context
+  const clankerMetadata: IClankerMetadata = {
+    description: formData.description || 'No description provided',
+    socialMediaUrls: [
+      ...(formData.telegramLink
+        ? [{ platform: 'telegram', url: formData.telegramLink }]
+        : []),
+      ...(formData.websiteLink
+        ? [{ platform: 'website', url: formData.websiteLink }]
+        : []),
+      ...(formData.xLink ? [{ platform: 'x', url: formData.xLink }] : []),
+      ...(formData.farcasterLink
+        ? [{ platform: 'farcaster', url: formData.farcasterLink }]
+        : []),
+    ],
+    auditUrls: [],
+  };
+
+  const clankerSocialContext: IClankerSocialContext = {
+    interface: 'YOUR_INTERFACE_HERE',
+    platform: userInfo?.farcaster ? 'farcaster' : userInfo?.twitter ? 'X' : '',
+    messageId: '',
+    id: userInfo?.farcaster?.fid
+      ? userInfo.farcaster.fid.toString()
+      : userInfo?.twitter?.username || '',
+  };
+
+  // Determine paired token
+  const pairAddress =
+    formData.pairedToken && formData.pairedToken.length > 0
+      ? (formData.pairedToken as `0x${string}`)
+      : WETH_ADDRESS;
+
+  // Calculate initial tick for price
+  // Note: This calculation assumes both tokens have standard 18 decimals.
```

Comment:
can we fix this to work for non standard decimals? maybe have it as an optional field that is passed in?